tests/validator_manager_pos_test.py

```diff
@@ -0,0 +1,68 @@
+import unittest, os
+
+from testcontainers.core.container import DockerContainer
+from substrateinterface import Keypair
+from app.lib.substrate import get_substrate_client
+from app.lib.validator_manager import get_validator_set, setup_pos_validator, get_validators_pending_addition, staking_chill, \
+    get_validators_pending_deletion
+from tests.test_utils import wait_for_http_ready
+from unittest import mock
+
+
+@mock.patch.dict(os.environ, {"TESTNET_MANAGER_CONSENSUS": "pos"})
+class ValidatorManagerTestPoS(unittest.TestCase):
+
+    def setUp(self):
+        # Start Alice validator
+        self.alice_validator = DockerContainer('parity/polkadot:latest')
+        self.alice_validator.with_command('--chain westend-local --validator --alice --unsafe-ws-external --rpc-cors=all')
+        self.alice_validator.with_exposed_ports(9944, 10333)
+        self.alice_validator.start()
+
+        self.alice_validator_rpc_ws_url = 'ws://{}:{}'.format(self.alice_validator.get_container_host_ip(),
+                                                              self.alice_validator.get_exposed_port(9944))
+
+        # Start Bob validator and connect it to Alice
+        self.bob_validator = DockerContainer('parity/polkadot:latest')
+        self.bob_validator.with_command("""
+            --chain westend-local --validator --bob --unsafe-ws-external --rpc-cors=all \
+            --bootnodes /ip4/127.0.0.1/tcp/{}/p2p/12D3KooWAvdwXzjmRpkHpz8PzUTaX1o23SdpgAWVyTGMSQ68QXK6
+        """.format(self.alice_validator.get_exposed_port(10333)))
+        self.bob_validator.with_exposed_ports(9933, 9944)
+        self.bob_validator.start()
+        self.bob_validator_http_url = 'http://{}:{}'.format(self.bob_validator.get_container_host_ip(),
+                                                            self.bob_validator.get_exposed_port(9933))
+        self.bob_validator_rpc_ws_url = 'ws://{}:{}'.format(self.bob_validator.get_container_host_ip(),
+                                                            self.bob_validator.get_exposed_port(9944))
+        self.alice_key = '0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a'
+
+        wait_for_http_ready(self.bob_validator_http_url + '/health')
+
+    def tearDown(self):
+        self.alice_validator.stop()
+        self.bob_validator.stop()
+
+    def test_get_validator_set(self):
+        validator_set = get_validator_set(self.bob_validator_rpc_ws_url)
+        print(validator_set)
+        self.assertTrue(validator_set, 'Validator set is present on the chain')
+
+    def test_register_validator(self):
```

Comment:
Please rename these functions to `test_register_pos_validator` `test_deregister_pos_validator`

Reply:
Done